[Caffe2] Add SVE128 vectorized<double> template layer

Nicoshev opened the pull request:
Summary: Add Vectorized<double> specialization for SVE128

Test Plan:
Ads Predictor

Rollback Plan:

Differential Revision: D80005963


cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

malfet commented on aten/src/ATen/native/DispatchStub.cpp:
```diff
@@ -32,26 +32,27 @@ static CPUCapability compute_cpu_capability() {
     if (envar == "vsx") {
       return CPUCapability::VSX;
-    }
+    } 
```
Trailing whitespace (and this change is not needed in general?

malfet commented on aten/src/ATen/native/DispatchStub.cpp:
```diff
@@ -32,26 +32,27 @@ static CPUCapability compute_cpu_capability() {
-#ifdef HAVE_ARM_BF16_CPU_DEFINITION
+    if (envar == "sve") {
+      if (sve_vl == 512) {
+        return CPUCapability::SVE512;
```
It talks about SVE128, why are we adding 512 here?

Nicoshev committed:
[Caffe2] Add SVE128 vectorized<int##bit##_t> template layers (#160326)

Summary:

Add Vectorized<T> SVE128 specialization for signed integer types

Test Plan: Predictor

Differential Revision: D79764906

Nicoshev committed:
[Caffe2] Add SVE128 vectorized<float16_t> template layer (#160325)

Summary:

Add Vectorized<float16_t> specialization for SVE128

Test Plan: Predictor

Reviewed By: mcfi

Differential Revision: D79385502

Nicoshev committed:
[Caffe2] Add SVE128 vectorized<double> template layer (#160329)

Summary:

Add Vectorized<double> specialization for SVE128

Test Plan: Ads Predictor

Differential Revision: D80005963

Nicoshev committed:
[Caffe2] Add NEON transpose routine (#164743)

Summary:

We are adding a custom NEON-based matrix transpose procedure, enabled for NEON and SVE128 translation layers.

Test Plan:
Ported the code to fbgemm and ran:

buck2 test mode/opt //deeplearning/fbgemm:TransposeTest

Differential Revision: D83978975

Nicoshev committed:
[Caffe2] Add SVE128 vectorized<T> template layers for unsigned integers (#160327)

Summary:

Add Vectorized<T> SVE128 specialization for unsigned integer types

Test Plan: Ads Predictor

Differential Revision: D79949491

Nicoshev committed:
[Caffe2] Add SVE128 vectorized<bfloat16_t> template layer (#160324)

Summary:

Add Vectorized<bfloat16_t> specialization for SVE128

Test Plan: Ads/Sigrid Predictor

Differential Revision: D79470607

Nicoshev committed:
[Caffe2] Add SVE128 vectorized<float> template layer (#160328)

Summary:

We are introducing the SVE128 vectorized<> layer.

Idea is to mix NEON and SVE to maximize performance on SVE128 cpus

This first diff adds vectorized<float>

Test Plan: Sigrid Predictor

Reviewed By: mcfi

Differential Revision: D79199881
